server/src/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -2188,7 +2188,13 @@ public UserVm updateVirtualMachine(UpdateVMCmd cmd) throws ResourceUnavailableEx
         }
 
         if (details != null && !details.isEmpty()) {
-            vmInstance.setDetails(details);
+            _vmDao.loadDetails(vmInstance);
```

Comment:
When do we need to remove detail? Even if needed then those values can be updated to get the desired effect. Many details are for internal purposes only and user is not even aware of them.